kythe/cxx/indexer/cxx/KytheIndexerMain.cc

```diff
@@ -119,6 +120,11 @@ int main(int argc, char* argv[]) {
   options.CreateWorklist = [](IndexerASTVisitor* indexer) {
     return IndexerWorklist::CreateDefaultWorklist(indexer);
   };
+  std::optional<FileHashRecorder> recorder;
+  if (!absl::GetFlag(FLAGS_record_hashes_file).empty()) {
+    recorder.emplace(absl::GetFlag(FLAGS_record_hashes_file));
+    options.HashRecorder = &*recorder;
```

Comment:
FYI (no change needed):  optional::emplace returns a reference to the emplaced object, so this can just be:
```
options.HashRecorder = &recorder.emplace(...);
```